src/main/java/org/owasp/validator/html/scan/MagicSAXFilter.java

```diff
@@ -363,8 +369,33 @@ public void startElement(QName element, XMLAttributes attributes, Augmentations
 					this.operations.push(Ops.FILTER);
 				} else {
 
-					if (isNofollowAnchors && "a".equals(element.localpart)) {
-						validattributes.addAttribute(makeSimpleQname("rel"), "CDATA", "nofollow");
+					if ("a".equals(element.localpart)) {
+						boolean addNofollow = isNofollowAnchors;
+						boolean addNoopenerAndNoreferrer = false;
+
+						if (isNoopenerAndNoreferrerAnchors) {
+							String targetValue = attributes.getValue("target");
+							if (targetValue != null && targetValue.equalsIgnoreCase("_blank")) {
+								addNoopenerAndNoreferrer = true;
+							}
+						}
```

Comment:
Combine these 2 lines.

Reply:
Same as the `ele.getAttributes().getNamedItem("target")` when we reach agreement I do the same resolution.